lib/loader.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const fs = require('fs');
+const fetch = require('node-fetch');
+const yaml = require('js-yaml');
+const resolver = require('./resolver.js');
+
+class ExtendableError extends Error {
+  constructor(message) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class OpenError extends ExtendableError {}
+class ReadError extends ExtendableError {}
+
+function readFileAsync(filename, encoding) {
```

Comment:
Copied from resolver, but resolver is going away in a future version as we ship the dependency to a new package. Let's just be ok with this for now.